reportek/core/models/reporting.py

```diff
@@ -426,31 +365,42 @@ def delete(self, *args, **kwargs):
                   f'{self.file.path}')
         super().delete(*args, **kwargs)
 
-    def get_download_url(self):
-        return reverse('api:envelope-file-download', kwargs={
-            'envelope_pk': self.envelope_id,
-            'pk': self.pk,
-        })
 
-    @classmethod
-    def get_or_create(cls, envelope, file_name):
-        """
-        Locates an `EnvelopeFile` based on `file_name`, or creates a new one.
-        Returns a tuple of the `EnvelopeFile` instance and a boolean indicating
-        if it is newly created.
-        """
-        is_new = True
-        try:
-            obj = cls.objects.get(
-                envelope=envelope,
-                name=file_name
-            )
-            is_new = False
+class EnvelopeOriginalFile(EnvelopeFileCommon):
+    """
+    Used to describe original non-XML files uploaded to the envelope
+    that are converted to and stored as XML EnvelopeFiles.
+    """
 
-        except cls.DoesNotExist:
-            obj = cls(envelope=envelope, name=file_name)
+    class Meta:
+        db_table = 'core_envelope_original_file'
+        unique_together = ('envelope', 'name')
 
-        return obj, is_new
+    # Overriding so get_download_url() works
+    _download_view_name = 'api:envelope-original-file-download'
+
+
+class EnvelopeFile(EnvelopeFileCommon):
+
+    class Meta:
```

Comment:
We should maybe have `Meta` in the concrete envelope file classes inherit from the ABC's `Meta`, in case we come up with some common meta-property (`unique_together` would qualify in this case, and see https://docs.djangoproject.com/en/1.11/topics/db/models/#meta-inheritance, where `ordering` is used as an example).